### PR TITLE
Set resources payload as a list

### LINKED ISSUE #7151: keycloak_authz_permission incorrectly sends a one element array as string
### Summary

When running:

```yaml
- name: Allow all users to list sessions
  community.general.keycloak_authz_permission:
    validate_certs: false
    name: allow-users-to-list-sessions
    state: present
    description: Allow users to list sessions
    permission_type: scope
    decision_strategy: AFFIRMATIVE
    scopes:
      - urn:ebo:CS:scopes:list
    resources:
      - Session
    policies:
      - all-users
    client_id: charging-stations
    realm: "{{ keycloak_realm }}"
    auth_keycloak_url: https://{{ ansible_default_ipv4.address }}:{{ keycloak_https_port }}/auth
    auth_username: admin
    auth_password: "{{ secret_keycloak_admin_password }}"
    auth_realm: master
```
It returns a 500, where java/keycloak prints:

```text
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.HashSet` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('34a556e1-95c0-482c-91df-4bccc865d356')
```

Which means it expects an array, but it got a string.

Looking at the python code [https://github.com/ansible-collections/community.general/blob/7721420388aa1d9cf7751fa250754d3419f3a2b1/plugins/modules/keycloak_authz_permission.py#L333](https://github.com/ansible-collections/community.general/blob/7721420388aa1d9cf7751fa250754d3419f3a2b1/plugins/modules/keycloak_authz_permission.py#L333):
```python
                payload['resources'] = r['_id']
```
I think this should be wrapped in an array.

    

### Issue Type

Bug Report

### Component Name

community.general.keycloak_authz_permission

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.15.3]
  config file = None
  configured module search path = ['/home/rgoussey/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/rgoussey/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/rgoussey/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/rgoussey/.local/bin/ansible
  python version = 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True

```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
Collection        Version
----------------- -------
community.general 7.3.0  

```

### Configuration

```console (paste below)

```

### OS / Environment

Ubuntu

### Steps to Reproduce

Tweak variables, and run against a running keycloak.

```yaml (paste below)
- name: Allow all users to list sessions
  community.general.keycloak_authz_permission:
    validate_certs: false
    name: allow-users-to-list-sessions
    state: present
    description: Allow users to list sessions
    permission_type: scope
    decision_strategy: AFFIRMATIVE
    scopes:
      - urn:ebo:CS:scopes:list
    resources:
      - Session
    policies:
      - all-users
    client_id: charging-stations
    realm: "{{ keycloak_realm }}"
    auth_keycloak_url: https://{{ ansible_default_ipv4.address }}:{{ keycloak_https_port }}/auth
    auth_username: admin
    auth_password: "{{ secret_keycloak_admin_password }}"
    auth_realm: master
```

### Expected Results

The task should use the correct serialization format.

### Actual Results

```console (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by fostermi**
##### SUMMARY
Fixes #7151 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
keycloak_authz_permission

##### ADDITIONAL INFORMATION

Before change, debug message was:
```
"msg": "Could not create update permission view.members.permission.group.f90a06ec-3d04-4e2e-8e43-e4faec6dc4a8 for client fc7011b1-ec2d-4e94-a53b-9f0a76b292c3 in realm testrealm: HTTP Error 500: Internal Server Error"
```
After change, task completes successfully:
```
PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

### CODE REVIEW

File: changelogs/fragments/7151-fix-keycloak_authz_permission-incorrect-resource-payload.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - keycloak_authz_permission - resource payload variable for scope-based permission was constructed as a string, when it needs to be a list, even for a single item. (https://github.com/ansible-collections/community.general/issues/7151).
```
Comment (felixfontein) [inline changelogs/fragments/7151-fix-keycloak_authz_permission-incorrect-resource-payload.yml:2]:
```suggestion
  - keycloak_authz_permission - resource payload variable for scope-based permission was constructed as a string, when it needs to be a list, even for a single item (https://github.com/ansible-collections/community.general/issues/7151).
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by mattock** [comment]
I ran the integration test suite with this patch applied on top of "main", Nothing breaks, at least not on Keycloak 20.0.2. According to the unofficial [API documentation](https://www.puppeteers.net/blog/keycloak-authorization-services-rest-api-paths-and-payload/) I created this is also in line how the Keycloak Admin UI constructs the JSON payload. 

Approval from my side :+1:

**Comment by fostermi** [comment]
Great! Is there anything else to do from my end?

**Comment by felixfontein** [comment]
@fostermi can you please accept the suggestion for the changelog fragment?

**Comment by felixfontein** [comment]
@fostermi thanks for fixing this!
@mattock thanks for reviewing this!

**Pull request merged**